src/adapter/src/catalog.rs

```diff
@@ -324,13 +324,19 @@ impl CatalogState {
         let index_matches =
             move |idx: &Index| idx.on == id && idx.compute_instance == compute_instance;
 
-        self.get_entry(&id)
-            .used_by()
-            .iter()
-            .filter_map(move |uses_id| match self.get_entry(uses_id).item() {
-                CatalogItem::Index(index) if index_matches(index) => Some((*uses_id, index)),
-                _ => None,
+        self.try_get_entry(&id)
+            .into_iter()
+            .map(move |e| {
+                e.used_by()
+                    .iter()
+                    .filter_map(move |uses_id| match self.get_entry(uses_id).item() {
```

Comment:
nit: `user_id` instead of `uses_id` is more obvous to me.

Reply:
`user_id` would be inaccurate since there is the possibility that the index is on a system source.